c7n/resources/opensearch.py

```diff
@@ -0,0 +1,122 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+from c7n.utils import local_session, type_schema
+from c7n.actions import BaseAction
+
+
+@resources.register('opensearch-serverless')
+class OpensearchServerless(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'opensearchserverless'
+        arn_type = 'arn'
+        enum_spec = ('list_collections', 'collectionSummaries[]', None)
+        name = "name"
+        id = "id"
+        universal_taggable = True
+        cfn_type = 'AWS::OpenSearchServerless::Collection'
+        arn = "arn"
+        permission_prefix = 'aoss'
+
+    def augment(self, resources):
```

Comment:
This function may not be needed if you use universal_augment i.e. `augment = universal_augment`